### PR TITLE
wrongly assumed add op had only 2 operands

### PR DESCRIPTION
**Pull request by swilly22**
#1612

### CODE REVIEW

File: tests/flow/test_bidirectional_traversals.py [hunk 1]
```diff
@@ -352,3 +352,22 @@ def test13_multiple_bidirectional_edges(self):
                            ['v3', 'v1'],
                            ['v3', 'v3']]
         self.env.assertEquals(actual_result.result_set, expected_result)
+
+    def test14_triple_edge_type(self):
```
Comment (jeffreylovitz) [inline tests/flow/test_bidirectional_traversals.py:356]: Put this test in a file like `test_relation_patterns.py`, as it does not test bidirectional traversals.

File: src/arithmetic/algebraic_expression/algebraic_expression_optimization.c [hunk 2]
```diff
@@ -112,48 +112,45 @@ static bool __AlgebraicExpression_MulOverAdd(AlgebraicExpression **root) {
...
-				C = r;
-
+				// lefthand side is addition
+				// (A + B + C) * D = (A * D) + (B * D) + (C * D)
+				B = r;
+				uint child_count = AlgebraicExpression_ChildCount(l);
+				for(uint i = 0; i < child_count; i++) {
+					A = _AlgebraicExpression_OperationRemoveDest(l);
+					AlgebraicExpression *mul = AlgebraicExpression_NewOperation(AL_EXP_MUL);
+					AlgebraicExpression_AddChild(mul, A);
```
Comment (jeffreylovitz) [inline src/arithmetic/algebraic_expression/algebraic_expression_optimization.c:132]: For consistency between the code paths, I'd put this line at 135 (to match 149-150).

**Review by jeffreylovitz — CHANGES_REQUESTED** [review]
Looks great! Minor changes requested.

File: src/arithmetic/algebraic_expression/algebraic_expression_optimization.c (hunk 2 above)
Comment (swilly22) [reply src/arithmetic/algebraic_expression/algebraic_expression_optimization.c:132]: can't, must maintain order matrix multiplication isn't commutative, the order in which we introduce children to `mul` is important.

**Pull request merged**